### PR TITLE
Fixes to JSONMonotoring (74X)

### PR DESCRIPTION
**Pull request by smorovic**
- fix calculating checksum on a wrong buffer (cause of spurious crashes)
- suppression of output when FastMonitoringService value is reset to 0

### CODE REVIEW

File: HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc [hunk 1]
```diff
@@ -573,6 +579,14 @@ TriggerJSONMonitoring::globalEndLuminosityBlockSummary(const edm::LuminosityBloc
     outHltJsnData<<result;
     outHltJsnData.close();
 
+    //HLT jsn entries
+    StringJ hltJsnFilelist_;
```
Comment (fwyzard) [inline HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc]: hi Srecko,
please do not use names ending with an underscore for non-member variables.

.A

File: HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc [hunk 2]
```diff
@@ -529,7 +531,14 @@ TriggerJSONMonitoring::endLuminosityBlockSummary(const edm::LuminosityBlock& iLu
 void
 TriggerJSONMonitoring::globalEndLuminosityBlockSummary(const edm::LuminosityBlock& iLumi, const edm::EventSetup& iSetup, const LuminosityBlockContext* iContext, hltJson::lumiVars* iSummary)
 {
-  if (iSummary->processed->value().at(0)!=0) {
+
+  unsigned int iLs  = iLumi.luminosityBlock();
+  unsigned int iRun = iLumi.run();
+
+  evf::FastMonitoringService * fms_ = (evf::FastMonitoringService *)(edm::Service<evf::MicroStateService>().operator->());
```
Comment (fwyzard) [inline HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc]: Please add a check for the availability of the FastMonitoringService or MicroStateService.
It is fine to print a LogError and/or disable some functionality if a service is not available, but it should not crash the whole program.

(actually, IMO *nothing* should ever crash the whole program, if a more graceful alternative is available)

File: HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc [hunk 3]
```diff
@@ -529,7 +531,14 @@ TriggerJSONMonitoring::endLuminosityBlockSummary(const edm::LuminosityBlock& iLu
...
+  bool writeFiles = fms_ ? fms_->getEventsProcessedForLumi(iLs)>0 : true;
```
Comment (fwyzard) [inline HLTrigger/JSONMonitoring/plugins/TriggerJSONMonitoring.cc]: Ops, apologies, ==fms_== is used only here, and there is a check - fooled by the name, I though it was a member available elsewhere.

**Comment by smorovic** [comment]
@fwyzard 
I changed all local variables to not use "_" (although this is Aram's code, I'm just fixing a bug...).
There is a check for the service, but I changed it a bit to be more readable.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by fwyzard** [comment]
hi @smorovic , I was looking only at the diff, but thanks for renaming the variables elsewhere, too.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**